runelite-client/src/main/java/net/runelite/client/plugins/questtab/QuestTabPlugin.java

```diff
@@ -0,0 +1,614 @@
+			case RESET:
+				configManager.setDefaultConfiguration(config, true);
+				break;
+		}
+	}
+
+	@Subscribe
+	public void onScriptEvent(ScriptCallbackEvent event)
+	{
+		String eventName = event.getEventName();
+		int[] intStack = client.getIntStack();
+		int intStackSize = client.getIntStackSize();
+		String[] stringStack = client.getStringStack();
+		int stringStackSize = client.getStringStackSize();
+
+		switch (eventName)
+		{
+			case "questListFilter":
+
+				if (intStackSize < 3)
+				{
+					break;
+				}
+
+				String questName = null;
+				Quest quest = null;
+
+				if (stringStackSize > 0)
+				{
+					questName = stringStack[stringStackSize - 1];
+					quest = Quest.getQuest(questName);
+				}
+
+				// order matters for these calls as filterOnProgress initializes a return value
+				// and the following calls maintain or update the value that was set
+				filterOnProgress(quest, questName, intStack, intStackSize);
+				filterOnQuestType(intStack, intStackSize);
+				filterOnQuestReqsMet(quest, questName, intStack, intStackSize);
+
+				// the second value on the int stack will be used as (true) once the script continues
+				intStack[intStackSize - 2] = 1;
+				break;
+			case "questListSetupDone":
+				filterList(intStack, intStackSize);
+				break;
+		}
+	}
+
+	private void filterOnProgress(Quest quest, String questName, int[] intStack, int intStackSize)
+	{
+		int progress = intStack[intStackSize - 3];
+
+		// progress == 2 means completed
+		if (progress == 2)
+		{
+			if (questName != null)
+			{
+				questStatus.put(quest, QuestStatus.COMPLETE);
+			}
+
+			intStack[intStackSize - 1] = config.hideCompleted() ? 1 : 0;
+		}
+		// progress == 0 means in progress
+		else if (progress == 0)
+		{
+			if (questName != null)
+			{
+				questStatus.put(quest, QuestStatus.IN_PROGRESS);
+			}
+
+			intStack[intStackSize - 1] = config.hideInProgress() ? 1 : 0;
+		}
+		// progress == 1 means not started
+		else if (progress == 1)
+		{
+			if (questName != null)
+			{
+				questStatus.put(quest, QuestStatus.NOT_STARTED);
+			}
+
+			intStack[intStackSize - 1] = config.hideNotStarted() ? 1 : 0;
+		}
+	}
+
+	private void filterOnQuestType(int[] intStack, int intStackSize)
+	{
+		int questListWidgetId = intStack[intStackSize - 2];
+
+		if (questListWidgetId == WidgetInfo.QUEST_LIST_FREE.getId())
+		{
+			intStack[intStackSize - 1] = config.hideFree() ? 1 : intStack[intStackSize - 1];
+		}
+		else if (questListWidgetId == WidgetInfo.QUEST_LIST_MEMBERS.getId())
+		{
+			intStack[intStackSize - 1] = config.hideMembers() ? 1 : intStack[intStackSize - 1];
+		}
+		else if (questListWidgetId == WidgetInfo.QUEST_LIST_MINIQUESTS.getId())
+		{
+			intStack[intStackSize - 1] = config.hideMiniquests() ? 1 : intStack[intStackSize - 1];
+		}
+	}
+
+	private void filterOnQuestReqsMet(Quest quest, String questName, int[] intStack, int intStackSize)
+	{
+		if (config.hideCantDo() && questName != null && !questStatus.get(quest).equals(QuestStatus.COMPLETE))

```

Comment:
Quests where you dont meet the requirements could have been started via a boost, and will be hidden. Should check for IN_PROGRESS also.